js/components/generation/select-sources-btn.html

```diff
@@ -8,7 +8,7 @@
 <!-- ko if: $component.shouldSuggestSelection() -->
     <select-sources-popup params="{
 		showModal: $component.isPopupShown,
-		sources: $component.sources,
+		sources: $component.sourceOptions,
```

Comment:
I would either leave the prev name or rename both - so to keep things consistent

Reply:
I'll rename to sourceOptions.

Reply:
Ok, once I stripped out that map() call, it made sense to keep those names consistent (so the param is `source` and the field is source.  You'll notice that i still kept a 'sourceOptions' field, tho, because we are still doing a transformation between the provided `sources` into a structure that contains a field (selected) that indicates of the user has selected the option or not.  Hopefully, tho, you'll see how the context of the object changes from source-> sourceOption and then the selections are returned as the array of sources (that i the exact same object that was initially passed in via params).

This may not be 100% clean but I think it's easier to follow this way. And, I think we can have a generic 'entity selector' component where you just need to provide the array of objects and the column definitions on how to read the values from the object and display as a column, and then you can have differnt implementations of a selector (cohort, featureanalysis, soures, ... anything that might need the user to select from a list of things).